src/cargo/ops/cargo_rustc/fingerprint.rs

```diff
@@ -289,3 +296,18 @@ fn filename(target: &Target) -> String {
     };
     format!("{}{}-{}", flavor, kind, target.get_name())
 }
+
+// The dep-info files emitted by the compiler all have their listed paths
```

Comment:
Possibly yeah, the source of the "problem" is that we call `rustc src/foo.rs` so the deps look like `src/foo.rs` and `src/bar.rs` instead of `/path/to/src/foo.rs`. I think that this mirrors gcc behavior, however. Either way, it's pretty easy to tweak in the future.